test/abilities/aroma_veil.test.ts

```diff
@@ -30,36 +29,37 @@ describe("Moves - Aroma Veil", () => {
       .enemyMoveset([MoveId.HEAL_BLOCK, MoveId.IMPRISON, MoveId.SPLASH])
       .enemySpecies(Species.SHUCKLE)
       .ability(Abilities.AROMA_VEIL)
-      .moveset([MoveId.GROWL]);
+      .moveset(MoveId.SPLASH);
   });
 
   it("Aroma Veil protects the Pokemon's side against most Move Restriction Battler Tags", async () => {
     await game.classicMode.startBattle([Species.REGIELEKI, Species.BULBASAUR]);
 
-    const party = game.scene.getPlayerParty()! as PlayerPokemon[];
+    const playerPokemon = game.scene.getPlayerField();
 
-    game.move.select(MoveId.GROWL);
-    game.move.select(MoveId.GROWL);
-    await game.forceEnemyMove(MoveId.HEAL_BLOCK);
+    game.move.select(MoveId.SPLASH);
+    game.move.select(MoveId.SPLASH);
+    await game.move.selectEnemyMove(MoveId.HEAL_BLOCK);
+    await game.move.selectEnemyMove(MoveId.SPLASH);
     await game.toNextTurn();
-    party.forEach((p) => {
+
+    playerPokemon.forEach((p) => {
       expect(p.getTag(BattlerTagType.HEAL_BLOCK)).toBeUndefined();
```

Comment:
```suggestion
      expect(p.getTag(BattlerTagType.HEAL_BLOCK)).toBeUndefined();
      expect(p.getLastXMoves()[0]?.result).toBe(MoveResult.SUCCESS);
```
To test that the Pokemon were actually allowed to use their moves due to Aroma Veil.

Reply:
Aroma Veil itself also needs a checkup soon...

https://github.com/user-attachments/assets/9f8c3b2f-19d4-4b27-91cd-acc528ceb0ae